core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -95,10 +98,17 @@
      */
     @BeforeClass
     public static void init() throws IOException {
+        Version version = Version.CURRENT;
+        // sometimes we need pre-1.3 index version to simulate disabled FieldNamesFieldMapper (e.g. for ExistQueryBuilder)
+        if (randomIntBetween(0, 5) == 0) {
+            version = Version.V_1_2_0;
+        }
```

Comment:
Yes, FieldNamesFieldMapper.isEnabled() depends on index version for bwc, switched on if Verison < Version.V_1_3_0. It's just here to covers one path in the ExistsQueryBuilder so far, so up until now I don't see a reason for having completey random versions.